storage/innobase/row/row0mysql.cc

```diff
@@ -2442,6 +2442,19 @@ row_insert_for_mysql(
 	const byte*		mysql_rec,
 	row_prebuilt_t*		prebuilt)
 {
+	struct CompressHeapCleaner {
```

Comment:
We followed the same style as prebuilt->blob_heap and I don't understand why we need a special treat for compress heap.

Reply:
There are two aspects here:
1. updating the row. Here the comments from Nitendra are valid (commit 59cb3d5f7da8e88d431f21e93ea60c094efdeb1b). In simple, we cannot free the uncompressed BLOB before storing it back. So the heap free should be done after storing the row (but that fix removed free completely)
2. Altering table (this ticket case). In this case, we are internally copying data between two tables row by row. As in 1 we removed heap free, we've got OOM.

So to satisfy 1 and 2 we have to free compression heap after row insertion